Crash on 6.0.1 with latest beta when changing interface

### Actual behaviour
When having the app in the background and changing from cell to wifi then restoring the app (either by widget or launcher), **sometimes** the app crashes. This happens approx. 3 out of 10 times.

### Expected behaviour
Should not crash when restoring

### Steps to reproduce
1. Connect to Wifi
2. Open App
3. Send App to background
4. Switch over to mobile and back to Wifi
5. Sometimes the app crashes initializing with the stack trace found below

### Can you reproduce the issue in demo mode?
Yes

### Environment data
#### Client
* Android version: 6.0.1
* Device model: Xiaomi Redmi 4 Prime
* Stock or customized ROM: Stock
* App version: 2.2.20-Beta
* Build flavor <!-- Play Store and GitHub = "Full", F-Droid = "Foss". If self-built and you haven't changed the flavor, it's Foss -->: Full
* Device language: German

#### Server
* Server version: 2.4.0~20180701105417-1
* Reverse Proxy: No
* Authentication method <!--(None, user and password, SSL Client certificate)-->: User+Pass

#### openHAB Cloud
* openHAB Cloud version: ?
* Self hosted cloud or myopenhab.org: myopenhab.org
* Authentication method <!--(None, user and password, SSL Client certificate)-->: User+Pass

### Logs
#### App log
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.openhab.habdroid.beta/org.openhab.habdroid.ui.OpenHABMainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'org.openhab.habdroid.util.AsyncHttpClient org.openhab.habdroid.core.connection.Connection.getAsyncHttpClient()' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2455)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2515)
	at android.app.ActivityThread.access$1000(ActivityThread.java:154)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1379)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:157)
	at android.app.ActivityThread.main(ActivityThread.java:5571)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:745)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:635)
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'org.openhab.habdroid.util.AsyncHttpClient org.openhab.habdroid.core.connection.Connection.getAsyncHttpClient()' on a null object reference
	at org.openhab.habdroid.ui.OpenHABMainActivity.loadSitemapIcon(OpenHABMainActivity.java:636)
	at org.openhab.habdroid.ui.OpenHABMainActivity.updateSitemapDrawerItems(OpenHABMainActivity.java:625)
	at org.openhab.habdroid.ui.OpenHABMainActivity.onCreate(OpenHABMainActivity.java:254)
	at android.app.Activity.performCreate(Activity.java:6357)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2408)
	... 9 more
```

#### openHAB Server log
No relevant output

#### Remarks
Maybe this is related to #953 
Would be great if this could be fixed before #549 as current play store build is not affected.

--- a/mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java
+++ b/mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java
@@ -250,7 +250,6 @@ public class OpenHABMainActivity extends AppCompatActivity implements
                 // via the fragment state restoration.
                 mController.recreateFragmentState();
             }
-            updateSitemapDrawerItems();
             if (savedInstanceState.getBoolean("isSitemapSelectionDialogShown")) {
                 showSitemapSelectionDialog();
             }

